### ISSUE TITLE
headless virtualbox option causes nixos-rebuild failure

### ISSUE THREAD

**Issue by chreekat** [NixOS/nixpkgs#157157]
## Issue description

When I try to rebuild a system with `virtualisation.virtualbox.host.headless = true`, I get the following message during nixos-rebuild:

```
building '/nix/store/6n0nkd1p2cpb0amb8krxrnd2s3qg16a7-ensure-all-wrappers-paths-exist.drv'...
Checking that Nix store paths of all wrapped programs exist... FAIL
The path /nix/store/rsxs7kp1yx03iq6vz4yyrxpsa50w345d-virtualbox-6.1.30/libexec/virtualbox/VBoxSDL does not exist!
Please, check the value of `security.wrappers."VBoxSDL".source`.
```

This problem went away when I changed the option to `false`.

### Steps to reproduce

(Untested outside of my own system)

Use the following in configuration.nix:

```
    # For local environments
    virtualbox = {
      host.enable = true;
      host.headless = true;
    };
```

Then, run `nixos-rebuild test`.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.93, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11.335526.6c4b9f1a2fd"`
 - channels(b): `"unstable-21.11pre318500.51bcdc4cdaa"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by Luflosi** [comment]
I'm having the same problem.
@chreekat please fix the typo in the title, so this is easier to find.

**Comment by chreekat** [comment]
Haha thanks for pointing that out!

**Comment by Luflosi** [comment]
After applying the patch
```diff
--- a/nixos/modules/virtualisation/virtualbox-host.nix
+++ b/nixos/modules/virtualisation/virtualbox-host.nix
@@ -110,9 +110,7 @@ in
       "VBoxNetAdpCtl"
       "VBoxNetDHCP"
       "VBoxNetNAT"
-      "VBoxSDL"
       "VBoxVolInfo"
-      "VirtualBoxVM"
     ]));
 
     users.groups.vboxusers.gid = config.ids.gids.vboxusers;
```
it seems to build fine.
I didn't actually run virtualbox yet, since I'm only using it for a BOINC project but haven't received any work units actually using virtualbox yet.
Can you test it please? If that works, I can make a PR that conditionally removes these two lines.

**Comment by chreekat** [comment]
@Luflosi unfortunately I'm not using VirtualBox anymore so I can't try this anymore. :/

**Comment by zhaofengli** [comment]
Should be fixed with #195141.
